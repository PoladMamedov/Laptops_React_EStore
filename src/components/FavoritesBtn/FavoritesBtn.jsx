export function FavoritesBtn(props) {
  return (
    <button data-testid="fav-btn" type="button" className="btn btn-light border-secondary position-relative pt-0">
      <img width={20} height={20} src="./img/star-active.png" alt="star" />
      <span className="position-absolute top-0 start-100 translate-middle badge rounded-pill bg-danger">
        {props.favItemsCount}
      </span>
    </button>
  );
}