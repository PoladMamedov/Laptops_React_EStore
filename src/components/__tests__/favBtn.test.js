import "@testing-library/jest-dom";
import { render, screen, cleanup } from "@testing-library/react";
import { FavoritesBtn } from "../FavoritesBtn/FavoritesBtn";
import renderer from "react-test-renderer";

afterEach(cleanup);

test("should redner favorites btn without text content", () => {
  render(<FavoritesBtn favItemsCount={0} />);
  expect(screen.getByTestId("fav-btn")).toBeInTheDocument();
  expect(screen.getByTestId("fav-btn")).toHaveTextContent("0");
});
test("should match the snaoshot", () => {
  const favoritesBtnSnap = renderer.create(<FavoritesBtn favItemsCount={0} />);
  expect(favoritesBtnSnap).toMatchSnapshot();
});
