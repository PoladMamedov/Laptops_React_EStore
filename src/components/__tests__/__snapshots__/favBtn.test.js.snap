// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snaoshot 1`] = `
<button
  className="btn btn-light border-secondary position-relative pt-0"
  data-testid="fav-btn"
  type="button"
>
  <img
    alt="star"
    height={20}
    src="./img/star-active.png"
    width={20}
  />
  <span
    className="position-absolute top-0 start-100 translate-middle badge rounded-pill bg-danger"
  >
    0
  </span>
</button>
`;
